firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -124,26 +129,25 @@ public static FirebaseAppDistribution getInstance(@NonNull FirebaseApp app) {
   @NonNull
   public Task<AppDistributionRelease> updateToLatestRelease() {
 
-    TaskCompletionSource<AppDistributionRelease> taskCompletionSource =
-        new TaskCompletionSource<>();
+    if (updateToLatestReleaseTaskCompletionSource != null
+        && !updateToLatestReleaseTaskCompletionSource.getTask().isComplete()) {
+      updateToLatestReleaseCancellationSource.cancel();
+    }
+
+    updateToLatestReleaseCancellationSource = new CancellationTokenSource();
+    updateToLatestReleaseTaskCompletionSource =
+        new TaskCompletionSource<>(updateToLatestReleaseCancellationSource.getToken());
 
+    // todo: when persistence is implemented and user already signed in, skip signInTester
     signInTester()
-        .addOnSuccessListener(
-            new OnSuccessListener<Void>() {
-              @Override
-              public void onSuccess(Void unused) {
-                taskCompletionSource.setResult(null);
-              }
-            })
+        .addOnSuccessListener(unused -> showUpdateDialogIfNewReleaseAvailable())
         .addOnFailureListener(
-            new OnFailureListener() {
-              @Override
-              public void onFailure(@NonNull Exception e) {
-                taskCompletionSource.setException(e);
-              }
-            });
+            e ->
```

Comment:
If you reset the exception here, the underlying AppDistributionException gets hidden. I would check if `e` is an instance of FirebaseAppDistribution and set the error as `e`. And maybe by default you can set the error to be `                    new FirebaseAppDistributionException(
                        Constants.ErrorMessages.AUTHENTICATION_ERROR, AUTHENTICATION_FAILURE)`